lecturework.html

```diff
@@ -0,0 +1,119 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="UTF-8" />
+    <title>Lesson 8 - Loops</title>
+  </head>
+
+  <body>
+    <div>
+      <h3>List of Products</h3>
+      <div>
+        <span>Iphone</span><br />
+        <span>Price: 900 €</span>
+        <input
+          type="button"
+          id="addPhoneToCart"
+          value="Add to shopping cart"
+        />
+      </div>
+      <br />
+      <div>
+        <span>Nintendo DS</span><br />
+        <span>Price: 1200 €</span>
+        <input
+          type="button"
+          id="addNintendoToCart"
+          value="Add to shopping car"
+        />
+      </div>
+      <br />
+      <div>
+        <span>MacBook Air</span><br />
+        <span>Price: 3200 €</span>
+        <input
+          type="button"
+          id="addMacbookToCart"
+          value="Add to shopping car"
+        />
+      </div>
+    </div>
+    <hr />
+    <div>
+      <h3>Shopping Cart</h3>
+      <input
+        type="button"
+        id="showShoppingCart"
+        value="Show Shopping Cart Products"
+      />
+      <div id="shoppingCart">
+        <!-- Display shopping cart products-->
+      </div>
+      <h3>
+        Total Price
+      </h3>
+
+      <span id="total">
+        <!-- show total price-->
+      </span>
+      <input type="button" id="totalPrice" value="Show total price" />
+    </div>
+
+    <script>
+
+      const phoneToCartEl = document.getElementById("addPhoneToCart");
+      const nintendoToCartEl = document.getElementById("addNintendoToCart");
+      const macbookToCartEl = document.getElementById("addMacbookToCart");
+      const showShoppingCartEl = document.getElementById("showShoppingCart");
+      const totalPriceEl = document.getElementById("total");
```

Comment:
Here you are getting the button with the wrong id, you should use `totalPrice`

Reply:
Carolyn, but above that I have this:
<span id="total">
        <!-- show total price-->
      </span>
Why do I want the button itself to by found by getElementById?
I will do it the way you suggested it, I just don't understand why it needs to be like that. Thanks.

Reply:
Good, seems working in general but I still have following problems:
(I'm actually going to compare it to your code or other students code as really confused what's wrong here):
1. in console, it doesn't show anything when I click on the product to be added to the Cart.
2. before, the display function for the Shopping cart worked nicely, now it stopped. Trying to fix that. 
I will try to update this doc here but because I did not get cloned to my computer before, i'm not sure i can just update it now or I will just upload a newer version of this. 
Sorry to bother you but i'm just trying to figure out why the heck the code is still not working. Thanks.

Reply:
1. if you want to see it in the console, you should add console.log(shoppingCart) in each function.
` function addIphoneToShoppingCart() {
        const product = {
          name: "Iphone",
          price: 900,
       }
       shoppingCart.push(product);
       console.log(shoppingCart) // this code allows to display the result in your console.
      }
`

Reply:
Thanks, added. Now it shows me adding the items. 
However, i have the last two problems:
1. shoppingCartContainer is not defined
2. Total sum sums up only the first two products I would click. Did I use a wrong += method for summing up?
Thanks!

Reply:
Here the solution:
**Issue #1:**
```
function showShoppingCartProducts() {
       // you need to defined first shoppingCartContainer outside the for loop, not inside.
       // the definition is just needed once.
        const shoppingCartContainer = document.getElementById('shoppingCart');
        shoppingCartContainer.innerHTML = '';

        for (let i = 0; i < shoppingCart.length; i++) {
          const productContainer = document.createElement('div');
          productContainer.innerHTML = `
          ${shoppingCart[i].name}<br>
          ${shoppingCart[i].price}
          `;
          shoppingCartContainer.appendChild(productContainer);
        }
      }
```

**Issue #2**
```
function showTotalPrice() {
        let total = 0;
        for (let i = 0; i < shoppingCart.length; i++) {
            // you need to use **i** to get the price of each product.  Before it was like
            // shoppingCart[0] -> that means that it just is getting the first product in your 
           // shopping cart, and summing up the same price.
            total += shoppingCart[i].price;
        }
        alert(total);
      }
```